polkadot/node/core/provisioner/src/lib.rs

```diff
@@ -646,15 +691,13 @@ async fn select_candidate_hashes_from_tracked(
 /// Should be called when prospective parachains are enabled.
 async fn request_backable_candidates(
 	availability_cores: &[CoreState],
+	elastic_scaling_mvp: bool,
 	bitfields: &[SignedAvailabilityBitfield],
 	relay_parent: Hash,
 	sender: &mut impl overseer::ProvisionerSenderTrait,
 ) -> Result<Vec<(CandidateHash, Hash)>, Error> {
 	let block_number = get_block_number_under_construction(relay_parent, sender).await?;
 
-	// These need to be ordered by core index. The runtime makes this assertion.
-	let mut selected_candidates: BTreeMap<CoreIndex, (CandidateHash, Hash)> = BTreeMap::new();
-
 	// Record which cores are scheduled for each paraid. Use a BTreeMap because
 	// we'll need to iterate through them.
 	let mut scheduled_cores: BTreeMap<ParaId, HashSet<CoreIndex>> = BTreeMap::new();
```

Comment:
Seems like this `HashSet` could be replaced with a counting integer?

Reply:
good catch! seems like it (it was an artifact from when we used to sort by core index)

Reply:
done